app/components/card_component.rb

```diff
@@ -1,5 +1,17 @@
 class CardComponent < ApplicationComponent
-  HEADER_VARIANTS = {default: "p-2 sm:p-4", no_padding: ""}
+  attr_accessor :media
+  FULL_WIDTH_16_BY_9 = [1290, 726]
+
+  def initialize(media: nil, **kwargs)
+    super(**kwargs)
+
+    self.media = media&.variant(resize_to_fill: FULL_WIDTH_16_BY_9)
```

Comment:
Why are we processing this here and now, rather than lazily? It presumes that when we initialize the `CardComponent` we know which variant to use?

Reply:
Yeah that's a good point, I think it would be better to do this in the view.